src/net/net.c

```diff
@@ -95,6 +95,48 @@ int net_dst_source_addr_get(struct sa *dst, struct sa *ip)
 }
 
 
+/**
+ * Check if connection from source to destination is possible
+ *
+ * @param src Source IP address
+ * @param dst Destination IP address
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int net_check_conn(const struct sa *src, const struct sa *dst)
+{
+	struct udp_sock *us;
+	struct sa srctmp;
+	struct sa dsttmp;
+	int port;
+	int err = 0;
+
+	if (!sa_isset(src, SA_ADDR) || !sa_isset(dst, SA_ADDR))
+		return EINVAL;
+
+	srctmp = *src;
+	dsttmp = *dst;
+	sa_cpy_scopeid(&dsttmp, src);
+	for (port = sa_port(src) + 3; port < sa_port(src) + 23; port++) {
+		err = udp_listen(&us, &srctmp, NULL, NULL);
+		if (!err)
+			break;
+
+		sa_set_port(&srctmp, port);
+	}
+
+	if (err)
+		return err;
+
+	err = udp_connect(us, &dsttmp);
```

Comment:
![Bildschirmfoto von 2021-06-08 13-00-22](https://user-images.githubusercontent.com/296762/121175010-e0428600-c85a-11eb-8625-435dbb04a9bd.png)

Its very unlikely but `make clang` complains about this.

Reply:
Found the `net_check_conn()` does not check anything useful. We can use `net_dst_source_addr_get()` instead, which retrieves a source address for which a given destination is reachable from.

I remove the commits that adds `net_check_conn()`.